config/opal_setup_wrappers.m4

```diff
@@ -541,7 +541,7 @@ AC_DEFUN([OPAL_SETUP_WRAPPER_FINAL],[
        OMPI_WRAPPER_FCFLAGS='-I${includedir}'" ${wrapper_extra_fcflags} ${with_wrapper_fcflags}"
        AS_IF([test -n "${OMPI_FC_MODULE_FLAG}"],
              [dnl deal with some interesting expansion behavior in OPAL_APPEND
-              wrapper_tmp_arg="${OMPI_FC_MODULE_FLAG}"'${libdir}'
+              wrapper_tmp_arg="${OMPI_FC_MODULE_FLAG} ${OMPI_FORTRAN_MODULEDIR}"
```

Comment:
@jsquyres After this change, what do you get in `mpifort-wrapper-data.txt`? Any chance this change will affect relocation of the install prefix?

Reply:
I'm wondering whether OPAL should learn to interpolate a new `${moddir}` variable, very much as it can handle `${libdir}` or `${includedir}` in `mpi*-wrapper-data.txt` files.

Reply:
> I'm wondering whether OPAL should learn to interpolate a new `${moddir}` variable, very much as it can handle `${libdir}` or `${includedir}` in `mpi*-wrapper-data.txt` files.

I think I see what you're saying.  If a user runs this configure:

```
./configure --with-mpi-moduledir=$HOME/bogus/include --prefix=$HOME/bogus ...
```

Then mpifort will show this:

```
$ mpifort --showme
gfortran -I/Users/jsquyres/bogus/include
```

And the datafile just has that hard-coded directory:

```
$ grep compiler_flags $prefix/share/openmpi/mpifort-wrapper-data.txt
compiler_flags=-I${includedir} -Wl,-flat_namespace  -I /Users/jsquyres/bogus/include
```

Meaning that if the user set `OPAL_PREFIX` to relocate their install, that 3rd argument won't follow.

In this case, it'll still just happen to be ok, because the 1st argument *will* follow `${includedir}`.  But that's a coincidence for this example.

So I see the issue -- but I'm not entirely sure how to address it.  The `OPAL_PREFIX` mechanism works because, by default, `$libdir` and friends are expressed relative to `$prefix`.  Do you envision the user specifying `--with-mpi-moduledir='$prefix/some_dir/this/is/where/i/want/modules'`?

Reply:
The primary motivation of the original request was just to install modules in `$includedir` rather than `$libdir`. So yes, at least in conda-forge, we want the moduledir to be a subpath within the install prefix. 

Maybe you can add some logic such that if if you pass `--with-mpi-moduledir=some/relative_path`, then that path is ultimately stored as `${prefix}/some/relative_path` in the `mpi*-wrapper-data.txt` files? Is that possible? Would that work?

Another option is to support something like  `--with-mpi-moduledir='${includedir}'`, and then the thing is 
expanded as appropriate in scripts, Makefiles, and `mpi*-wrapper-data.txt` files.

Reply:
I guess the entire `OPAL_PREFIX` scheme is dependent upon everything being a subdirectory under `$prefix` -- e.g., if you `./configure --prefix/opt/openmpi --libdir=/usr/lib ...`, then setting `OPAL_PREFIX` won't affect `libdir` at all.

I just did a test, and the existing code does support this kind of scheme, even without `${moddir}`.  For example:

```
$ ./configure --prefix=/tmp/bogus '--with-mpi-moduledir=${includedir}/modules' ...
$ make install
...
$ /tmp/bogus/bin/mpifort --showme
gfortran -I/tmp/bogus/include -Wl,-flat_namespace -Wl,-commons,use_dylibs -I /tmp/bogus/include/modules -L/tmp/bogus/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi
$ mv /tmp/bogus /tmp/bogus2
$ export OPAL_PREFIX=/tmp/bogus2
$ /tmp/bogus2/bin/mpiforst --showme
gfortran -I/tmp/bogus2/include -Wl,-flat_namespace -Wl,-commons,use_dylibs -I /tmp/bogus2/include/modules -L/tmp/bogus2/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi
```

Is this sufficient, or are you looking for something more?

I could certainly add something in the docs about how using `OPAL_PREFIX` with `--with-mpi-moduledir` (and/or any of the other directory override CLI options) would require careful use of `'${blah}'` (where `blah` is one of the recognized names like `prefix`, `libdir`, `includedir`, ... etc.).

Reply:
> Is this sufficient, or are you looking for something more?

I think this is perfectly sufficient for the needs of conda-forge.

BTW, looking at your output, why is there whitespace between `-I` and the module dir ? Let me take a look...

Reply:
Maybe you should remove the blank space as below? Looking at the previous code using `libdir`, there was no blank space.

```suggestion
              wrapper_tmp_arg="${OMPI_FC_MODULE_FLAG}${OMPI_FORTRAN_MODULEDIR}"
```

Reply:
Good catch; will fix.  I'll add some docs to this commit in a little bit.

Reply:
@jsquyres Sorry, just to double check... If you configure using `--with-mpi-moduledir='${includedir}/modules'`, what do you get written in `mpifort-wrapper-data.txt` ? Do yuo see (literally) `-I${includedir}/modules` in there?

Reply:
@dalcinl Yes, I literally see `-I${includedir}/modules`, and therefore it gets the same `OPAL_PREFIX` treatment.

Reply:
@dalcinl This is the v5.0.x cherry pick, so I have to make the changes on `main` first, and then cherry-pick them over here.  I opened PR https://github.com/open-mpi/ompi/pull/12667 to remove the errant space and add the new docs.